sklearn/ensemble/_forest.py

```diff
@@ -516,7 +516,10 @@ def fit(self, X, y, sample_weight=None):
             n_more_estimators > 0 or not hasattr(self, "oob_score_")
         ):
             y_type = type_of_target(y)
-            if y_type in ("multiclass-multioutput", "unknown"):
+            if y_type == "unknown" or (
+                y_type == "multiclass-multioutput"
+                and isinstance(self, (RandomForestClassifier, ExtraTreesClassifier))
+            ):
```

Comment:
Let's use the `_estimator_type` here instead of the isinstance.

```suggestion
            if y_type == "unknown" or (
                self._estimator_type == "classifier"
                and y_type == "multiclass-multioutput"
            )):
```